### ISSUE TITLE
Can't return focus to java app.

### ISSUE THREAD

**Issue by kostafey** [stumpwm/stumpwm#235]
Can't return focus to java app (mucommander in my case). The same old story. But I can't find the correct solution. Cat you provide any advice?

``` lisp
(defun handle-mucommander-subwindow-close (window)
  (if  (equal (window-class window)
              "com-mucommander-Launcher")
       (let ((w (car
                 (stumpwm::find-matching-windows
                  '(:class "com-mucommander-Launcher")
                  'all-groups 'all-screens))))
         ;; useless ;(
         (focus-window w t)
         ;; works, but ugly
         (ratwarp 1000 800)
         (ratclick))))

(add-hook *destroy-window-hook* 'handle-mucommander-subwindow-close)
```

**Comment by jorams** [comment]
What version of StumpWM are you using, and what should I do to reproduce this?
I just downloaded mucommander and it isn't giving me any focus issues, but perhaps I'm not trying hard enough.

**Comment by kostafey** [comment]
Stumpwn from git:
```
commit b63be557ec52123b1d4b24aa4a537103ebac3dfd
Author: David Bjergaard <dbjergaard@gmail.com>
Date:   Sun Oct 4 15:14:46 2015 +0100

    Makefile rule tweak to extract filenames from stumpwm.asd
```

You can just try to copy some file (F5), you'll see confirmation dialog and than cancel it (Esc). The focus wil not return to mucommander.

**Comment by jorams** [comment]
I did a bit of research and I *think* I found the source of the problem:

In focus-window there are essentially two approaches for focusing a window. The first is used when the window has `WM_TAKE_FOCUS` set and `*current-event-time*` is set. The third in all other cases. Now, normally the mucommander window would want focus using the first approach, and it does, but when the event causing the focus is `:destroy-notify`, `*current-event-time*` will not be set and the second approach is trieid, which is ineffective.

I'm not sure how this should be solved. Simply passing a bogus time doesn't work and the ICCCM requires it to be a valid timestamp.

**Comment by dbjergaard** [comment]
Can we set *current-event-time* from destroy-notify, setting it to the time when
destroy-notify is called?

**Comment by kostafey** [comment]
Probably there is a way to set a focus manually on destroy afer some sleeping period?

**Comment by jorams** [comment]
@dbjergaard The problem is getting a valid server timestamp. I read somewhere else that GTK does/did this by causing some bogus event to trigger and then catching its timestamp, but I don't currently have time to look into that.

**Comment by jorams** [comment]
I'm pretty sure this was fixed by #297.
